sql/src/test/java/io/crate/jobs/transport/NodeDisconnectJobMonitorServiceTest.java

```diff
@@ -59,23 +62,51 @@ public void testOnNodeDisconnectedKillsJobOriginatingFromThatNode() throws Excep
         JobExecutionContext context = jobContextService.createContext(builder);
 
         ThreadPool threadPool = mock(ThreadPool.class);
-        when(threadPool.schedule(any(TimeValue.class), anyString(), any(Runnable.class))).thenAnswer(new Answer<Object>() {
-            @Override
-            public Object answer(InvocationOnMock invocation) throws Throwable {
-                ((Runnable) invocation.getArguments()[2]).run();
-                return null;
-            }
+        when(threadPool.schedule(any(TimeValue.class), anyString(), any(Runnable.class))).thenAnswer((Answer<Object>) invocation -> {
+            ((Runnable) invocation.getArguments()[2]).run();
+            return null;
         });
 
         NodeDisconnectJobMonitorService monitorService = new NodeDisconnectJobMonitorService(
             Settings.EMPTY,
             threadPool,
             jobContextService,
-            mock(TransportService.class));
+            mock(TransportService.class),
+            mock(TransportKillJobsNodeAction.class));
 
         monitorService.onNodeDisconnected(new DiscoveryNode("noop_id", DummyTransportAddress.INSTANCE, Version.CURRENT));
 
         expectedException.expect(ContextMissingException.class);
         jobContextService.getContext(context.jobId());
     }
+
+    @Test
+    public void testOnParticipatingNodeDisconnectedKillsJob() throws Exception {
+        JobContextService jobContextService = new JobContextService(
+            Settings.EMPTY, new NoopClusterService(), mock(StatsTables.class));
+
+        DiscoveryNode coordinator_node = new DiscoveryNode("coordinator_node_id", DummyTransportAddress.INSTANCE, Version.CURRENT);
+        DiscoveryNode data_node = new DiscoveryNode("data_node_id", DummyTransportAddress.INSTANCE, Version.CURRENT);
+
+        DiscoveryNodes discoveryNodes = DiscoveryNodes.builder()
+            .localNodeId("coordinator_node_id")
+            .put(coordinator_node)
+            .put(data_node).build();
+
+        jobContextService.newBuilder(UUID.randomUUID(), coordinator_node.getId(), Arrays.asList(coordinator_node.getId(), data_node.getId()));
+        TransportKillJobsNodeAction killAction = mock(TransportKillJobsNodeAction.class);
+
+        NodeDisconnectJobMonitorService monitorService = new NodeDisconnectJobMonitorService(
+            Settings.EMPTY,
+            mock(ThreadPool.class),
+            jobContextService,
+            mock(TransportService.class),
+            killAction);
+
+        monitorService.onNodeDisconnected(discoveryNodes.get("data_node_id"));
+        verify(killAction, times(1)).broadcast(
```

Comment:
@mfussenegger not sure if this is the best way to test the broadcasting of the kill request. Maybe you know some better technique ;)